R/sem.R

```diff
@@ -540,13 +568,97 @@ checkLavaanModel <- function(model, availableVars) {
       "boot",               gettext("bootstrap (Bollen-Stine) probability value")
     )
     testname <- LUT[test == tolower(LUT$option), "name"][[1]]
-    ftext <- gettextf("Model tests based on %s.", testname)
-    fittab$addFootnote(message = ftext)
+    if (length(semResults) == 1)
+      ftext <- gettextf("Baseline tests based on %s.", testname)
+    if (length(semResults) > 1)
+      ftext <- gettextf("Baseline tests based on %s. Difference tests based on a function of two standard test-statistics.", testname)
+
+    fnote <- paste0(fnote, ftext)
+
   }
 
   if (options$estimator %in% c("dwls", "gls", "wls", "uls")) {
-    fittab$addFootnote(message = gettext("The AIC, BIC and additional information criteria are only available with ML-type estimators"))
+    fnote <- paste0(fnote, gettext("The AIC, BIC and additional information criteria are only available with ML-type estimators"))
+  }
+
+  if (options[["group"]] != "") {
+
+    groupNames <- semResults[[1]]@Data@group.label
+    models <- rep(rownames(lrt$value), each = length(groupNames))
+    lavopts <- .semOptionsToLavOptions(options, dataset)
+    lavopts[["group"]] <- NULL
+    results_grouped <- list()
+    k = 1
+    for (i in 1:length(options[["models"]])) {
+      syntax   <- lavaan::parTable(semResults[[i]])
+      for (group in seq_along(groupNames)) {
+        lav_args <- lavopts
+        syntax_group <- syntax[syntax$group == group,]
+        equalityConstraints <- syntax[syntax$op == "==",]
+        if (nrow(equalityConstraints) > 0) {
+          for (j in 1:nrow(equalityConstraints)) {
+            if(equalityConstraints[j, "lhs"] %in% syntax_group[, "plabel"]) {
+              syntax_group <- rbind(syntax_group, equalityConstraints[j,])
+            }
+          }
+        }
+        syntax_group[["group"]] <- syntax_group[["block"]] <- rep(1, nrow(syntax_group))
+        lav_args[["model"]] <- syntax_group
+        dataset_group <- dataset[dataset[[options[["group"]]]] == groupNames[group],]
+        lav_args[["data"]]  <- dataset_group
+        fit_group <- try(do.call(lavaan::lavaan, lav_args))
```

Comment:
@Kucharssim, this code does not work when there are equality constraints, which is not surprising. Lorenzo fits the model for the subgroup only, but with invariance testing equality constraints are between the two groups, e.g., the loadings of group1 are equal to the loadings of group2. It seems to me that the equality constraints do not matter for the fit of each group, so we should estimate the model with free parameters unless some the constraints are not between groups but within a group. I wonder how we would find distinguish here though....